charon/src/translate/translate_ctx.rs

```diff
@@ -437,12 +437,64 @@ impl<'tcx, 'ctx> TranslateCtx<'tcx, 'ctx> {
         let opaque = attributes
             .iter()
             .any(|attr| attr == "charon::opaque" || attr == "aeneas::opaque");
+        let rename = {
+            let filter_attribute: Vec<&str> = attributes
+                .iter()
+                .filter_map(|str| {
+                    str.strip_prefix("charon::rename(")
+                        .or(str.strip_prefix("aeneas::rename("))
+                        .and_then(|str| str.strip_suffix(")"))
+                })
+                .collect();
+            if let [str] = filter_attribute.as_slice() {
+                let rename = str
+                    .strip_prefix("\"")
+                    .and_then(|str| str.strip_suffix("\""));
+                if let Some(str) = rename {
+                    if !str.is_empty() {
+                        let first_char_alphabetic = str
+                            .chars()
+                            .nth(0)
+                            .expect("Attribute `rename` should not be empty")
+                            .is_alphabetic();
+                        let is_identifier = first_char_alphabetic
+                            && str
+                                .chars()
+                                .all(|c| c.is_alphanumeric() || c == '_' || c == '-');
+                        if !is_identifier {
+                            self.span_err(span, "Attribute `rename` should only contains alphanumeric characters, `_` and `-` and should start with a letter");
+                            None
+                        } else {
+                            Some(rename.unwrap().to_string())
+                        }
+                    } else {
+                        self.span_err(span, "Attribute `rename` should not be empty");
```

Comment:
Don't use `span_err` directly, we have macros that do the right thing and register errors. Probably you want to return a `Result<.., Error>` and use the macro `register_error_or_panic`

Reply:
actually no, it would be silly to have to skip the whole item if an attribute is malformed. returning `ItemMeta` is fine, bug we need to register the error so charon knows there was one

Reply:
oh, `span_err` does that already, nevermind